src/EditorFeatures/Core.Wpf/BackgroundWorkIndicator/BackgroundWorkIndicatorScope.cs

```diff
@@ -3,88 +3,49 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Threading;
 using Microsoft.VisualStudio.Utilities;
-using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Editor.BackgroundWorkIndicator;
-
-internal partial class WpfBackgroundWorkIndicatorFactory
+namespace Microsoft.CodeAnalysis.Editor.BackgroundWorkIndicator
 {
-    /// <summary>
-    /// Implementation of an <see cref="IUIThreadOperationScope"/> for the background work indicator. Allows for
-    /// features to create nested work with descriptions/progress that will update the all-up indicator tool-tip
-    /// shown to the user.
-    /// </summary>
-    private sealed class BackgroundWorkIndicatorScope : IUIThreadOperationScope, IProgress<ProgressInfo>
+    internal partial class WpfBackgroundWorkIndicatorFactory
     {
-        private readonly BackgroundWorkIndicatorContext _context;
-
-        // Mutable state of this scope.  Can be mutated by a client, at which point we'll ask our owning context to
-        // update the tooltip accordingly.
-
-        private string _description;
-        private ProgressInfo _progressInfo;
-
-        public IUIThreadOperationContext Context => _context;
-        public IProgress<ProgressInfo> Progress => this;
-
-        public BackgroundWorkIndicatorScope(
-            BackgroundWorkIndicatorContext indicator, string description)
-        {
-            _context = indicator;
-            _description = description;
-        }
-
-        /// <summary>
-        /// Retrieves a threadsafe snapshot of our data for our owning context to use to build the tooltip ui.
-        /// </summary>
-        public (string description, ProgressInfo progressInfo) ReadData_MustBeCalledUnderLock()
-        {
-            Contract.ThrowIfFalse(Monitor.IsEntered(_context.Gate));
-            return (_description, _progressInfo);
-        }
-
-        /// <summary>
-        /// On disposal, just remove ourselves from our parent context.  It will update the UI accordingly.
-        /// </summary>
-        void IDisposable.Dispose()
-            => _context.RemoveScope(this);
-
-        bool IUIThreadOperationScope.AllowCancellation
+        private class BackgroundWorkIndicatorScope : IUIThreadOperationScope
```

Comment:
do we need our own impl here?  or is this because the provided wait-indicator has no way to create scopes?

Reply:
The implementation from platform doesn't inherit from `IUIThreadOperationScope`, just `IDisposable`. This gets into the "Do we want an abstraction"  question. It would require more changes to get rid of that abstraction, but may be worth it.

Reply:
That's... interesting.  How do teh scopes work?  @kayle can you point to the docs on that?

Reply:
They're very basic right now. You can create/dispose them and update their descriptions. The current presentation will show the description of the last (not-disposed) scope:
```
    /// <summary>
    /// Represents a single scope in the context of executing a potentially long running operation.
    /// </summary>
    /// <remarks>Members can be invoked from any thread.</remarks>
    public abstract class BackgroundWorkOperationScope : IDisposable
    {
        internal BackgroundWorkOperationScope() { } // Prevent external implementations

        /// <summary>
        /// Updates the description of the operation scope.
        /// </summary>
        public abstract string Description { set; }

        /// <summary>
        /// Marks the operation scope as completed
        /// </summary>
        public abstract void Dispose();
    }
```

I do expect we'll add progress reporting in the future, but possibly not for operations owned by a TextView. We will likely not mix the IUIThreadOperationScope with the BackgroundWorkOperationScope or any future entry points on IBackgroundWorkIndicatorService.

Reply:
>We will likely not mix the IUIThreadOperationScope with the BackgroundWorkOperationScope

I do find that very very odd.  From my perspective, these are just a different, lightweight, view on the same model and controller.  Having a different API with disparate functionality seems odd (esp. as the functionality seems relevant to all views).

Reply:
When we discussed this previously, we acknowledged that not everything in `IUIThreadOperationScope` made sense to move forward with. I believe it was `AllowCancellation` specifically that didn't make sense for BWI

Reply:
>  I believe it was AllowCancellation specifically that didn't make sense for BWI

Hrmm... that seems odd to me.  We have operations (like applying a solution-change) that need to switch over to being non-cancellable once they start actually making the change.  This way the user doesn't end up in some sort of partial-application state.

Reply:
There is a way to disallow cancellation, I think it was just meant to not be a property. I would have to digup e-mail for exact reasoning